packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -0,0 +1,386 @@
+   * @param account Address of the account
+   * @param trustedIssuers Array of n issuers whose identifier mappings will be used
+   * @param maxIdentifiers Limit the number of identifiers that will be returned
+   * @return Array (length <= maxIdentifiers) of identifiers
+   * @dev Adds identifier info to the arrays in order of provided trustedIssuers
+   * @dev Expectation that only one attestation exists per (identifier, issuer, account)
+   */
+
+  // TODO ASv2 consider also returning an array with counts of identifiers per issuer
+
+  function lookupIdentifiersByAddress(
+    address account,
+    address[] memory trustedIssuers,
+    uint256 maxIdentifiers
+  ) public view returns (bytes32[] memory) {
+    // Same as for the other lookup, preallocate and then trim for now
+    uint256 currIndex = 0;
+    bytes32[] memory identifiers = new bytes32[](maxIdentifiers);
+
+    for (uint256 i = 0; i < trustedIssuers.length; i = i.add(1)) {
+      address trustedIssuer = trustedIssuers[i];
+      uint256 numIssuersForAddress = addressToIdentifiers[account][trustedIssuer].length;
+      for (uint256 j = 0; j < numIssuersForAddress; j = j.add(1)) {
+        // Iterate through the list of identifiers
+        if (currIndex < maxIdentifiers) {
+          bytes32 identifier = addressToIdentifiers[account][trustedIssuer][j];
+          // Check if the mapping was produced by a revoked signer
+          // solhint-disable-next-line max-line-length
+          OwnershipAttestation[] memory attestationsForIssuer = identifierToAddresses[identifier][trustedIssuer];
+          for (uint256 k = 0; k < attestationsForIssuer.length; k = k.add(1)) {
+            OwnershipAttestation memory attestation = attestationsForIssuer[k];
+            // (identifier, account, issuer) tuples should be unique
+            if (attestation.account == account && !revokedSigners[attestation.signer]) {
+              identifiers[currIndex] = identifier;
+              currIndex = currIndex.add(1);
+              break;
+            }
+          }
+        } else {
+          break;
+        }
+      }
+    }
+    if (currIndex < maxIdentifiers) {
+      // Allocate and fill properly-sized array
+      bytes32[] memory trimmedIdentifiers = new bytes32[](currIndex);
+      for (uint256 i = 0; i < currIndex; i = i.add(1)) {
+        trimmedIdentifiers[i] = identifiers[i];
+      }
+      return trimmedIdentifiers;
+    } else {
+      return identifiers;
+    }
+  }
+
+  // TODO do we want to restrict permissions, or should anyone
+  // with a valid signature be able to register an attestation?
+  modifier isValidUser(address issuer, address account) {
+    require(
+      msg.sender == account ||
+        msg.sender == issuer ||
+        getAccounts().attestationSignerToAccount(msg.sender) == issuer,
+      "User does not have permission to perform this action"
+    );
+    require(!revokedSigners[msg.sender], "User has been revoked ");
```

Comment:
Can accounts be revoked? Based on the map naming I had assumed this was only for issuer signer addresses, but this seems to indicate that is not true?

Reply:
I believe this should be "Signer has been revoked," cc @isabellewei

Reply:
Let me clarify this. What I notice here is that if a client account (who is an attestation consumer and not an issuer), could potentially be added to the `revokedSigners` mapping. If they are, then they would no longer pass this check and the user would be unable to register or delete attestations. It seems like an edge case, but maybe this is intentional? Should this maybe be changed such that revocation is only checked if the sender is an issuer?

Reply:
There was a question of whether we should only allow authorized `AttestationSigner` signers to be added to the `revokedSigner` mapping, but we weren't sure if we should allow for a scenario where a signer has been unauthorized in the `Accounts` contract but the issuer still wants to revoke them here.

Reply:
ahh misunderstood earlier (though I do still think the text should be `Signer has been revoked` not `user`)

>  this maybe be changed such that revocation is only checked if the sender is an issuer

- yes I agree -- in theory we want it to still be possible for accounts to have attestations about them and to be able to submit those, even if they are revoked signers (though I do think this is a reasonably unlikely edge case?); @isabellewei we should incorporate that into the require condition

> There was a question of whether we should only allow authorized AttestationSigner signers to be added to the revokedSigners mapping, but we weren't sure if we should allow for a scenario where a signer has been unauthorized in the Accounts contract but the issuer still wants to revoke them here.
I think this is a separate thing. We still should allow issuers to revoke themselves (i.e. in the case of smart contract as issuer where there isn't a signer, we should make sure it's possible for an account to revoke itself as an issuer). We just shouldn't mix the account vs. issuer/signer permissions in this case -- an address can be both a revoked signer and account simultaneously, but it's only "invalid" in a situation if it's supposed to be serving the role of signer, and should be "valid" if it's serving the role of account.